docs/static/doc-for-plugin.asciidoc

```diff
@@ -0,0 +1,184 @@
+want them to see. Incorrect link formatting or duplicate links can break the
+documentation build. Let's not do that. 
+
+===== Link to content in the same file
+
+Use angle brackets to format links to content in the same asciidoc file. 
+
+*Example*
+
+This link:
+[source,txt]
+-----
+<<plugins-{type}s-{plugin}-config_models>>
+-----
+
+Points to this heading in the same file:
+
+[source,txt]
+----------------------------------
+[id="plugins-{type}s-{plugin}-config_models"]
+==== Configuration models
+----------------------------------
+
+
+===== Link to content in the Logstash Reference Guide
+
+Use external link syntax for links that point to documentation for other plugins or content in the Logstash Reference Guide.
+
+*Examples*
+[source,txt]
+-----
+{logstash-ref}/plugins-codecs-multiline.html[Multiline codec plugin]
+-----
+
+[source,txt]
+-----
+{logstash-ref}/getting-started-with-logstash.html
+-----
+
+===== Link text
+
+If you don't specify link text, the URL is used as the link text.
+
+
+*Examples*
+
+If you want your link to display as {logstash-ref}/getting-started-with-logstash.html, use this format:
+[source,txt]
+-----
+{logstash-ref}/getting-started-with-logstash.html
+-----
+
+
+If you want your link to display as {logstash-ref}/getting-started-with-logstash.html[Getting Started with Logstash], use this format:
+[source,txt]
+-----
+{logstash-ref}/getting-started-with-logstash.html[Getting Started with Logstash]
+-----
+
+
+===== Link to data type descriptions
+
+We make an exception for links that point to data type descriptions,
+such as `<<boolean,boolean>>`, because they are used so frequently. 
+We have a cleanup step in the conversion script that converts the links to the
+correct syntax. 
+
+
+[[format-code]]
+==== Code samples
+
+We all love code samples. Asciidoc supports code blocks and config examples.
+To include Ruby code, use the asciidoc `[source,ruby]` directive.
+
+Note that the hashmarks (#) are present to make the example render correctly.
+Don't include the hashmarks in your asciidoc file.
+
+[source,txt]
+[subs="attributes"]
+----------------------------------
+# [source,ruby]
+# -----
+# match => {
+#  "field1" => "value1"
+#  "field2" => "value2"
+#  ...
+# }
+# -----
+----------------------------------
+
+The sample above (with hashmarks removed) renders in the documentation like this:
+[source,ruby]
+----------------------------------
+match => {
+  "field1" => "value1"
+  "field2" => "value2"
+  ...
+}
+----------------------------------
+
+==== Where's my doc?
+
+Plugin documentation goes through several steps before it gets published in the 
+{lsplugindocs}[Logstash Versioned Plugin Reference] and the {logstash-ref}/index.html[Logstash Reference].
+
+Here's an overview of the workflow:
+
+* Be sure that you have signed the CLI and have all necessary approvals and sign offs.
+* Merge the pull request for your plugin (including the `index.asciidoc` file, the `changelog.md` file, and the gemspec).
```

Comment:
Wondering if you need to be more direct so that contributors do update the changelog and bump the version?